test/index.js

```diff
@@ -330,6 +352,15 @@ describe('KeyringController', function () {
         expect(keyring.wallets).toHaveLength(1);
       });
     });
+    it('add serialized keyring to unsupported_keyrings array if keyring type is not known', async function () {
+      mockEncryptor.encrypt(password, [
+        { type: 'Ledger Keyring', data: 'DUMMY' },
+      ]);
+      await keyringController.setLocked();
+      const keyrings = await keyringController.unlockKeyrings(password);
+      expect(keyrings).toHaveLength(0);
+      expect(keyringController.unsupported_keyrings).toHaveLength(1);
```

Comment:
This test would pass if the code added anything to `unsupported_keyrings`. To be more accurate, what do you think about:

``` suggestion
      expect(keyringController.unsupported_keyrings).toStrictEqual(keyrings);
```